src/routines/executeQuery.ts

```diff
@@ -346,6 +347,26 @@ export const executeQuery = async (
 
   // Stream does not have `rows` in the result object and all rows are already transformed.
   if (result.rows) {
+    if (slonikSqlRename.zodObject && slonikSqlRename.zodObject?._def?.typeName !== 'ZodAny') {
+      for (const row of result.rows) {
+        const validationResult = slonikSqlRename.zodObject.safeParse(row);
```

Comment:
Zod objects can transform data, so this might lead to unexpected behaviour. e.g. an object like `z.object({ foo: z.string().transform(s => s.split(',')) })` has an "output" type of `{ foo: string[] }` but since all this does is validate (it ignores the successful parse result), the row type at runtime will be `{ foo: string }`, and the code will expect `{ foo: string[] }`. An `interceptor.transformRow` value could be dynamically added to properly handle this, maybe?